src/poetry/utils/dependency_specification.py

```diff
@@ -64,7 +64,7 @@ def _parse_dependency_specification_url(
     if not (url_parsed.scheme and url_parsed.netloc):
         return None
 
-    if url_parsed.scheme in ["git+https", "git+ssh"]:
+    if url_parsed.scheme in ["git+http", "git+https", "git+ssh"]:
```

Comment:
Let's make this (and the list below for URL deps) a file-level constant set (e.g. `GIT_URL_SCHEMES = {}`, `URL_SCHEMES = {}`).

Reply:
It looks like simply `startswith("git+")` would be enough:
https://github.com/python-poetry/poetry-core/blob/e72d3e497c7d9298bdabae179e8c58e58a82903b/src/poetry/core/packages/dependency.py#L425

Reply:
`git+` is not good enough as we don't support `git+file://` -- the original approach of a set is much better.

Reply:
Yes, it's much better, but I don't think it makes much sense as long as the underlying library doesn't do in that way. I think it should surely be addressed, but I guess it's not what should be done in this PR.

Reply:
The rationale is that it's even better off sharing the same logic between poetry and poetry-core, and that should be done in the separate PR. I don't want to dig this even further, like factoring out the list never used in other places.

Reply:
This is still the correct place to make this check; the code in poetry-core is generic URL parsing and not opinionated on what URLs are valid for a Python package vs. Git in general.

I am fine to move that logic into poetry-core, but I am not fine with removing the check from Poetry before that code is ready. Even though this is a fairly off-the-beaten path concern, making things more permissive and restricting it again later is not a viable approach.